dungeon/test/manual/quizquestion/WizardQuizTest.java

```diff
@@ -0,0 +1,135 @@
+
+public class WizardQuizTest {
+
+
+    public static QuizQuestion singleChoiceDummy() {
+        QuizQuestion question =
+            new QuizQuestion(
+                QuizQuestion.QuizType.SINGLE_CHOICE,
+                QuizQuestion.QuizContentType.TEXT,
+                "Was ist kein Ziel von Refactoring?");
+        question.addAnswer(QuizQuestion.QuizContentType.TEXT, "Lesbarkeit von Code verbessern");
+        question.addAnswer(
+            QuizQuestion.QuizContentType.TEXT, "Verständlichkeit von Code verbessern");
+        question.addAnswer(
+            QuizQuestion.QuizContentType.TEXT, "Wartbarkeit von Code verbessern");
+        question.addAnswer(QuizQuestion.QuizContentType.TEXT, "Fehler im Code ausmerzen");
+        return question;
+    }
+
+
+    private static final Consumer<Entity> wizardInteractionCallback =
+        entity -> {
+            Task quest = entity.fetch(TaskComponent.class).orElseThrow().task();
+            // build GUI with its callback
+            QuizUI.showQuizDialog((QuizQuestion) quest, (Entity wizard)->{
+            System.out.println("DEBUG WIZARD: 58" );
+                // TODO answer=
+                return (textDialog, id) -> {
+                    if (Objects.equals(id, core.hud.UITools.DEFAULT_DIALOG_CONFIRM)) {
+                        SnapshotArray<Actor> children = textDialog.getContentTable().getChildren();
+                        // find the answersection .added a name to it for easier search
+                        var answerSection = (VerticalGroup)children.select((actor) -> Objects.equals(actor.getName(), ANSWERS_GROUP_NAME)).iterator().next();
+                        System.out.println("DEBUG WIZARD: 65" + answerSection);
+                        // do some magic
+                        var answerText = switch (((QuizQuestion)quest).type()){
+                            case SINGLE_CHOICE -> getSingleChoiceText(answerSection);
+                            case MULTIPLE_CHOICE -> ""; //TODO ???
+                            case FREETEXT -> getFreeTextAnswer(answerSection);
+                        };
+                        System.out.println("DEBUG WIZARD: 70" );
+                        //antwort steht in answerText
+                        //UITools.generateNewTextDialog(answerText,"Ok","The answer was");
+                        System.out.println(answerText);
+                        Game.removeEntity(entity);
+                        return true;
+                    }
+                    return false;
+
+                };
+            });
+
+        };
+
+    private static String getSingleChoiceText(VerticalGroup answerSection) {
+        return ((VerticalGroup)
+            ((ScrollPane) answerSection.getChildren().get(0))
+                .getChildren().get(0)).getChildren().select((x) -> x instanceof CheckBox checkbox && checkbox.isChecked())
+            .iterator()
+            .next() instanceof CheckBox checked ? checked.getName() : "No Selection";
```

Comment:
```suggestion
            .next() instanceof CheckBox checked ? checked.getText().toString() : "No Selection";
```
Und hier habe ich vergessen den text anstelle von dem Namen zu verwenden.

Reply:
Top ich danke dir.
Das schön machen kann ich übernehmen.